linux/mcompile/keymap/filesystem.cpp

```diff
@@ -0,0 +1,188 @@
+
+#ifdef __EMSCRIPTEN__
+#include <emscripten.h>
+#include <emscripten/bind.h>
+#endif
+
+#include <cassert>
+#include <algorithm>
+#include <iostream>
+#include "filesystem.h"
+
+#ifdef _MSC_VER
+#include <io.h>
+#else
+#include <unistd.h>
+#endif
+
+//#define _DEBUG_FOPEN  1
+
+#ifdef __EMSCRIPTEN__
```

Comment:
We don't need any emscripten support for mcompile, because it will only run native -- so we can delete all emscriptn/wasm-related code blocks.

Reply:
OK-addressed  -> #11887
filesystem is removed from mcompile-linux 
function Open_File is moved to the module mc_kmxfile